store/store.go

```diff
@@ -2480,3 +2482,36 @@ func (s *Store) ConnectivityCheck() (status map[string]bool, err error) {
 
 	return status, nil
 }
+
+func (s *Store) CreateCohorts(ctx context.Context, snaps []string) (map[string]string, error) {
+	jsonData, err := json.Marshal(map[string][]string{"snaps": snaps})
+	if err != nil {
+		return nil, err
+	}
+
+	u := s.endpointURL(cohortsEndpPath, nil)
+	reqOptions := &requestOptions{
+		Method:   "POST",
+		URL:      u,
+		APILevel: apiV2Endps,
+		Data:     jsonData,
+	}
+
+	var remote struct {
+		CohortKeys map[string]string `json:"cohort-keys"`
+	}
+	resp, err := s.retryRequestDecodeJSON(ctx, reqOptions, nil, &remote, nil)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case 200:
+		// OK
+	case 404:
+		return nil, ErrSnapNotFound
```

Comment:
Just wondering, would be nice to say which particular snap was not found when you pass multiple.. but that I guess would need changes in the store.

Reply:
we'd have to parse the error response; compare:
```
~$ http https://api.staging.snapcraft.io/v2/cohorts snaps:='["test-snapd-tools-foo", "xyzzy-moopmoop"]'
HTTP/1.1 404 NOT FOUND
Content-Length: 189
Content-Type: application/json
Date: Mon, 29 Apr 2019 09:20:02 GMT
Server: gunicorn/19.7.1
Snap-Store-Version: 13
X-Request-Id: D9A9015EDB1C0A324E7501BB5CC6C1C211766
X-VCS-Revision: 165f1b8
X-View-Name: snapdevicegw.webapi_cohort.create_cohort

{
    "error-list": [
        {
            "code": "resource-not-found",
            "message": "No snap with name 'xyzzy-moopmoop' found"
        },
        {
            "code": "resource-not-found",
            "message": "No snap with name 'test-snapd-tools-foo' found"
        }
    ]
}

~$ http https://api.staging.snapcraft.io/v2/cohorts snaps:='["test-snapd-tools", "xyzzy-moopmoop"]'
HTTP/1.1 404 NOT FOUND
Content-Length: 100
Content-Type: application/json
Date: Mon, 29 Apr 2019 09:20:39 GMT
Server: gunicorn/19.7.1
Snap-Store-Version: 13
X-Request-Id: D9A9015EDB200A324E7501BB5CC6C1E61176D
X-VCS-Revision: 165f1b8
X-View-Name: snapdevicegw.webapi_cohort.create_cohort

{
    "error-list": [
        {
            "code": "resource-not-found",
            "message": "No snap with name 'xyzzy-moopmoop' found"
        }
    ]
}
```

Reply:
I see, thanks for the example, it would be nice to do at some point IMHO.